### PR TITLE
Revamp existing secrets

### LINKED ISSUE #196: Docs: Example of keeping secret values in k8s
I'd really appreciate some documentation showing the required values to implement all secrets in k8s. I'm finding it is a bit more complex than the README expects. When implementing this for Enterprise, I keep finding more things I need to add and things that are conflicting throwing me into a guess and check loop that I'd consider a UX bug that could be fixed by clearer documentation. 

Thank you

### PR DESCRIPTION
**Pull request by Btodhunter**
fixes #196 #182 #52

### CODE REVIEW

**Review by HN23 — COMMENTED** [review]
can we add a changelog to the readme

**Review by HN23 — CHANGES_REQUESTED** [review]
can we add a changelog to the readme

File: stable/anchore-engine/templates/enterprise_feeds_secret.yaml [hunk 1]
```diff
@@ -14,7 +14,16 @@ metadata:
     {{- end }}
 type: Opaque
 stringData:
+  ANCHORE_FEEDS_DB_NAME: {{ index .Values "anchore-feeds-db" "postgresDatabase" | quote }}  
+  ANCHORE_FEEDS_DB_USER: {{ index .Values "anchore-feeds-db" "postgresUser"  | quote }}
   ANCHORE_FEEDS_DB_PASSWORD: {{ index .Values "anchore-feeds-db" "postgresPassword" | quote }}
+  {{- if and (index .Values "anchore-feeds-db" "externalEndpoint") (not (index .Values "anchore-feeds-db" "enabled")) }}
```
Comment (HN23) [inline stable/anchore-engine/templates/enterprise_feeds_secret.yaml:20]: nit: can we make this consistent with engine where it uses the helper template

**Review by HN23 — DISMISSED** [review]
lgtm

File: stable/anchore-engine/README.md [hunk 2]
```diff
@@ -609,29 +619,63 @@ anchoreApi:
...
+apiVersion: v1
+kind: Secret
+metadata:
+  name: anchore-engine-env
+type: Opaque
+stringData:
+  ANCHORE_ADMIN_PASSWORD: foobar1234
+  ANCHORE_DB_NAME: anchore
+  ANCHORE_DB_USER: anchoreengine
+  ANCHORE_DB_HOST: anchore-postgresql:5432
```
Comment (HN23) [inline stable/anchore-engine/README.md:646]: NIT: people might copy/paste this and create the secret but deploy the platform using a different name. I think we should put a note next to the <hosts> to let them know what has to change with the deployment name if not naming the helm deployment 'anchore'

**Review by HN23 — APPROVED** [review]
lgtm

File: stable/anchore-engine/README.md (hunk 2 above)
Comment (Btodhunter) [reply stable/anchore-engine/README.md:646]: Good observation! I'll add a comment to the secret explaining that these secrets will only work as-is if they're using helm deployed postgres/redis with the default values & a release name of `anchore`.

**Review by HN23 — APPROVED** [review]
lgtm

**Comment by flickerfly** [comment]
This'll be a big enhancement and clean some stuff up a bunch! Thank you!

**Comment by flickerfly** [comment]
I just put 1.21.0 into production and it is so much cleaner to deal with existingSecrets using the default names, pulled out a bunch of stuff from values.yaml I didn't need anymore and added one line to replace it all. I really appreciated the improved documentation on the process as well. Thank you @Btodhunter! I'm sure this'll promote others more pain averse then I was 8mo ago to make more secure deployments.

**Pull request merged**